pkg/controller/postgressnapshot/postgressnapshot_controller.go

```diff
@@ -237,3 +245,52 @@ func (r *ReconcilePostgresSnapshot) createSnapshot(ctx context.Context, rdsSvc r
 	r.logger.Info(msg)
 	return croType.PhaseInProgress, croType.StatusMessage(msg), nil
 }
+
+func buildPostgresSnapshotInfoMetricLabels(cr *integreatlyv1alpha1.PostgresSnapshot, clusterID, snapshotName string) map[string]string {
+	labels := buildPostgresSnapshotGenericMetricLabels(cr, clusterID, snapshotName)
+	if len(string(cr.Status.Phase)) != 0 {
+		labels["statusPhase"] = string(cr.Status.Phase)
+		return labels
+	}
+	return labels
+}
+
+func buildPostgresSnapshotGenericMetricLabels(cr *integreatlyv1alpha1.PostgresSnapshot, clusterID, snapshotName string) map[string]string {
+	labels := map[string]string{}
+	labels["clusterID"] = clusterID
+	labels["resourceID"] = cr.Name
+	labels["namespace"] = cr.Namespace
+	labels["instanceID"] = snapshotName
+	labels["productName"] = cr.Labels["productName"]
+	labels["strategy"] = postgresProviderName
+	labels["statusPhase"] = ""
```

Comment:
Do we need `statusPhase` here in the generic labels? Technically these should always stay generic

Reply:
yes its needed here. run into issues with cardinality if not present when you attempt to add statusPhase later in buildPostgresSnapshotInfoMetricLabels

Reply:
no your right don't need it here.